packages/dds/tree/README.md

```diff
@@ -163,12 +176,44 @@ Each change is processed in two ways:
 -   the change is accumulated in a `ProgressiveEditBuilder` which will be used to create/encode the actual edit to send to Fluid.
 
 Once the command ends, the transaction is rolled back leaving the forest in a clean state.
-Then if the command did not error, a [`change-set`](./src/changeset/README.md) is created from the `ProgressiveEditBuilder`, which is encoded into a Fluid Op.
+Then if the command did not error, a [`changeset`](./src/changeset/README.md) is created from the `ProgressiveEditBuilder`, which is encoded into a Fluid Op.
 The checkout then rebases the op if any Ops came in while the transaction was pending (only possible for async transactions or if the checkout was behind due to it being async for some reason).
 Finally the checkout sends the op to `shared-tree-core` which submits it to Fluid.
 This submission results in the op becoming a local op, which `shared-tree-core` creates a delta for.
 This delta goes to the indexes, resulting in the ForestIndex and thus checkouts getting updated,
 as well as anything else subscribing to deltas.
 
+This shows completion of a transaction.
+Not shown are the rollback or changes to forest (and the resulting invalidation) and AnchorSet,
+then the updating of them with the final version of the edit.
+In the common case this can be skipped (since they cancel out).
+Also not shown is the (also usually unneeded) step of rebasing the changeset before storing it and sending it to the service.
+
+```mermaid
+graph RL;
+    command["App's command callback"]--"completes transaction"-->transaction
```

Comment:
There is a "completes transaction" arrow here and in the following line which is a bit confusing for me.

Reply:
When the application's command completes the transaction (returns in the case of a synchronous transaction), that is information flow from the command callback to the transaction which kicks of this whole process causing the transaction to submit the changes. I'm not sure how to more clearly capture that. Any suggestions?

Reply:
Hmm maybe either replacing the first with "trigger transaction completion" or the second with something along the lines of "finalize edits into transaction"? I think it'll help as long as it's clear in what component the transaction is actually being completed.

Reply:
I have made some more changes. Maybe its more clear now.